vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/RenderingManager.java

```diff
@@ -0,0 +1,315 @@
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.model.display.ISpaceLinkDisplay;
+import edu.asu.diging.vspace.core.model.impl.BranchingPoint;
+import edu.asu.diging.vspace.core.model.impl.SequenceHistory;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.IExternalLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.IRenderingManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.ISpaceDisplayManager;
+import edu.asu.diging.vspace.core.services.ISpaceLinkManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Service
+public class RenderingManager implements IRenderingManager {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+       
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private SlideManager slideManager;
+
+    @Autowired
+    private ISequenceManager sequenceManager;
+    
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+    
+    @Autowired
+    private SpringTemplateEngine springTemplateEngine;
+
+    @Autowired
+    private ISpaceDisplayManager spaceDisplayManager;
+
+    @Autowired
+    private IExhibitionManager exhibitManager;
+
+    @Autowired
+    private IModuleLinkManager moduleLinkManager;
+
+    @Autowired
+    private ISpaceLinkManager spaceLinkManager;
+
+    @Autowired
+    private IExternalLinkManager externalLinkManager;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private IStorageManager storageManager;
+    
+    private final String IMAGES_FOLDER_NAME = "images";
+       
+    private final String SPACE_TEMPLATE_DOWNLOAD_API = "exhibition/downloads/spaceDownloadTemplate";
+    
+    private final String SLIDE_TEMPLATE_DOWNLOAD_API = "exhibition/downloads/slideDownloadTemplate";
+    
+    private final String DOWNLOAD_FILE_EXTENSION = ".html"; 
+    
+    /**
+     * 
+     * Download given space and related modules into exhibitionFolderPath
+     * @param space                            the space object 
+     * @param exhibitionFolderName             the folder name of the exhibition where space contents will be stored           
+     * @param sequenceHistory                  the sequence history object having the history of sequences                        
+     * @throws FileStorageException 
+     */
+    public void downloadSpace(Space space, String exhibitionFolderName,  SequenceHistory sequenceHistory) throws FileStorageException {
```

Comment:
this method is not downloading anything, is it? Just creating a snapshot/rendering the pages?